src/controllers/settings_controller.js

```diff
@@ -160,21 +161,31 @@ SettingsController.post('/subscribe', function (request, response) {
 });
 
 SettingsController.post('/skiptunnel', function (request, response) {
```

Comment:
Ideally SettingsController would be a PromiseRouter then you could write this with async/await and be more explicit about the lifetime of the route handler.